### PR TITLE
AArch64: pmap emulate bits

### PR DESCRIPTION
**Pull request by pj1031999**
Port #954 for AArch64.

### CODE REVIEW

**Comment by cahirwpz** [comment]
Please align your implementation with _MIPS_ counterpart.

File: sys/aarch64/pmap.c [hunk 1]
```diff
@@ -206,8 +214,8 @@ static paddr_t pmap_alloc_pde(pmap_t *pmap, vaddr_t vaddr) {
   return pg->paddr;
 }
 
-static pte_t make_pte(paddr_t pa, vm_prot_t prot, unsigned flags) {
-  pte_t pte = pa | vm_prot_map[prot];
+static pte_t make_pte(paddr_t pa, pte_t prot, unsigned flags) {
```
Comment (bkjg) [inline sys/aarch64/pmap.c:217]: Is this change really necessary?

File: sys/aarch64/pmap.c [hunk 2]
```diff
@@ -286,7 +294,7 @@ void pmap_kenter(vaddr_t va, paddr_t pa, vm_prot_t prot, unsigned flags) {
 
   klog("Enter unmanaged mapping from %p to %p", va, pa);
 
-  pte_t pte = make_pte(pa, prot, flags);
+  pte_t pte = make_pte(pa, vm_prot_map[prot], flags);
```
Comment (bkjg) [inline sys/aarch64/pmap.c:297]: Why cannot we do it the way we did it earlier? To take the appropriate prots in the make_pte function?
